tests/callbacks/test_finetuning_callback.py

```diff
@@ -342,3 +342,59 @@ def forward(self, x):
     # conv0.weight, conv0.bias, bn0.weight, bn0.bias
     # conv1.weight, conv1.bias, bn1.weight, bn1.bias
     assert len(encoder_params) == 8
+
+
+def test_parent_module_w_param_model():
```

Comment:
>  but I at least initially have broken the test out separately for clarity. 

It is fine to replace the one above as you are just adding checks! So please do

Reply:
Great idea, thanks. I've added another commit that replaces test_deep_nested_model with the new test and changes the test name to be more general as I suspect there may be other minor checks we will want to add for more complex models that may not warrant separate tests. Also updated changelog. Thanks for the great framework! Looking forward to contributing in any meager way I can.